source/bsd/kqueue_event_loop.c

```diff
@@ -798,7 +799,7 @@ static void s_event_thread_main(void *user_data) {
     struct kqueue_loop *impl = event_loop->impl_data;
 
     /* set thread id to the event-loop's thread. */
-    aws_atomic_store_int(&impl->thread_id, (size_t)aws_thread_current_thread_id());
+    aws_atomic_store_ptr(&impl->thread_id, &impl->thread.thread_id);
```

Comment:
This could also use `aws_atomic_exchange_ptr` with an expected `NULL` value as an additional check that `run` and `stop; wait` calls are interleaved properly.

Reply:
Suggestion, but not implemented.